hack/update-generated-code.sh

```diff
@@ -27,7 +27,7 @@ CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-ge
 #                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
 #                  instead of the $GOPATH directly. For normal projects this can be dropped.
 bash ${CODEGEN_PKG}/generate-groups.sh "deepcopy,client,informer,lister" \
-  github.com/kubernetes-csi/external-snapshotter/pkg/client github.com/kubernetes-csi/external-snapshotter/pkg/apis \
+  github.com/kubernetes-csi/external-snapshotter/client github.com/kubernetes-csi/external-snapshotter/client/v2/apis \
```

Comment:
You missed the 2nd `v2` on this line:

github.com/kubernetes-csi/external-snapshotter/client/v2/apis -> github.com/kubernetes-csi/external-snapshotter/client/apis

Reply:
@Kartik494 I think it is okay just to keep v2 in the path
`github.com/kubernetes-csi/external-snapshotter/client/v2 github.com/kubernetes-csi/external-snapshotter/client/v2/apis \`

After running ./hack/update-generated-code.sh from root, move all generated files from client\v2 to client, and then delete v2 folder.

Make sure that you document the procedure in README.

Reply:
@xing-yang Yes, since we created the go.mod for client/v2 and move apis under client folder, so the path to the apis should be "github.com/kubernetes-csi/external-snapshotter/client/v2/apis".

> After running ./hack/update-generated-code.sh from root, move all generated files from client\v2 to client, and then delete v2 folder.
> 
> Make sure that you document the procedure in README so that it can be repeated by others.

Yes, when i run the script with `github.com/kubernetes-csi/external-snapshotter/client`  only then i get the output as : 
 
Generating clientset for volumesnapshot:v1beta1 at github.com/kubernetes-csi/external-snapshotter/client/clientset
Generating listers for volumesnapshot:v1beta1 at github.com/kubernetes-csi/external-snapshotter/client/listers
Generating informers for volumesnapshot:v1beta1 at github.com/kubernetes-csi/external-snapshotter/client/informers
 
and i have updated this in README as well.

could you please clarify on this `After running ./hack/update-generated-code.sh from root, move all generated files from client\v2 to client, and then delete v2 folder`, as i have deleted the v2 folder earlier

Reply:
I have downloaded your PR and this is what I see.  Right after the PR is downloaded and before running any script:
```
~/go/src/github.com/kubernetes-csi/external-snapshotter# ls -l client
total 60
drwxr-xr-x 3 root root  4096 Aug  5 13:58 apis
drwxr-xr-x 3 root root  4096 Aug  5 13:58 clientset
drwxr-xr-x 3 root root  4096 Aug  5 13:58 config
-rw-r--r-- 1 root root   310 Aug  5 13:58 go.mod
-rw-r--r-- 1 root root 33192 Aug  5 13:58 go.sum
drwxr-xr-x 3 root root  4096 Aug  5 13:58 informers
drwxr-xr-x 3 root root  4096 Aug  5 13:58 listers
```
Then run the script:
```
~/go/src/github.com/kubernetes-csi/external-snapshotter# ./hack/update-generated-code.sh
Generating deepcopy funcs
Generating clientset for volumesnapshot:v1beta1 at github.com/kubernetes-csi/external-snapshotter/client/clientset
Generating listers for volumesnapshot:v1beta1 at github.com/kubernetes-csi/external-snapshotter/client/listers
Generating informers for volumesnapshot:v1beta1 at github.com/kubernetes-csi/external-snapshotter/client/informers
```
Here is what I see after running the script.  Note that a "v2" folder is automatically generated by the script.
```
~/go/src/github.com/kubernetes-csi/external-snapshotter# ls -l client
total 64
drwxr-xr-x 3 root root  4096 Aug  5 13:58 apis
drwxr-xr-x 3 root root  4096 Aug  5 13:58 clientset
drwxr-xr-x 3 root root  4096 Aug  5 13:58 config
-rw-r--r-- 1 root root   310 Aug  5 13:58 go.mod
-rw-r--r-- 1 root root 33192 Aug  5 13:58 go.sum
drwxr-xr-x 3 root root  4096 Aug  5 13:58 informers
drwxr-xr-x 3 root root  4096 Aug  5 13:58 listers
drwxr-xr-x 3 root root  4096 Aug  6 12:34 v2
```
A new zz_generated.deepcopy.go is generated under client/v2:
```
~/go/src/github.com/kubernetes-csi/external-snapshotter/client/v2/apis/volumesnapshot/v1beta1# ls -l
total 16
-rw-r--r-- 1 root root 12449 Aug  6 12:34 zz_generated.deepcopy.go
```

Here is my suggestion.  In the hack/update-generated-code.sh, after running the line that generates code, copy the file generated under client/v2/apis to client/apis:
```
bash ${CODEGEN_PKG}/generate-groups.sh "deepcopy,client,informer,lister" \
  github.com/kubernetes-csi/external-snapshotter/client github.com/kubernetes-csi/external-snapshotter/client/v2/apis \
  volumesnapshot:v1beta1 \
  --go-header-file ${SCRIPT_ROOT}/hack/boilerplate.go.txt

+# Copy generated file from client/v2 to client folder
+cp -rf client/v2/apis/volumesnapshot/v1beta1 client/apis/volumesnapshot/
+# Remove client/v2 folder
+rm -rf client/v2
```

Reply:
Hi @xing-yang when i run this in my enviorment i didn't get v2 folder created under client  after running the script

`~/go/src/github.com/external-snapshotter$ ls -l client

total 60
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 apis
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 clientset
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 config
-rw-r--r-- 1 necuser necuser   310 Aug  6 18:59 go.mod
-rw-r--r-- 1 necuser necuser 33192 Aug  6 18:59 go.sum
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 informers
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 listers
`

When i run the script
 
`./hack/update-generated-code.sh 
Generating deepcopy funcs

Generating clientset for volumesnapshot:v1beta1 at github.com/kubernetes-csi/external-snapshotter/client/clientset

Generating listers for volumesnapshot:v1beta1 at github.com/kubernetes-csi/external-snapshotter/client/listers

Generating informers for volumesnapshot:v1beta1 at github.com/kubernetes-csi/external-snapshotter/client/informers
`
After the script i got the output as :
 

`/go/src/github.com/external-snapshotter$ ls -l client
total 60
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 apis
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 clientset
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 config
-rw-r--r-- 1 necuser necuser   310 Aug  6 18:59 go.mod
-rw-r--r-- 1 necuser necuser 33192 Aug  6 18:59 go.sum
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 informers
drwxr-xr-x 3 necuser necuser  4096 Aug  6 18:59 listers

`
I am not sure on why it is showing different results, i am looking into this

Reply:
@Kartik494 how about this issue?  Can you reproduce what I've seen?

Reply:
@xing-yang yes i have run this in my enviorment but i am not finding any folder v2 is generated in client.

Reply:
I tried your PR again and I still get a v2 generated under client.